code/game/gamemodes/gang/gang_items.dm

```diff
@@ -75,6 +77,185 @@
+			final_guard = FALSE
+			reinforce(FALSE)
+
+/obj/machinery/gang/backup/proc/reinforce(var/repeat = TRUE)
+	if(!src)
+		return
+	var/we = 0
+	var/rival = 0
+	var/cooldown = 0
+	queue = list()
+	for(var/datum/gang/baddies in SSticker.mode.gangs)
+		if(baddies == G)
+			for(var/datum/mind/M in G.gangsters)
+				if(M.current.stat == DEAD)
+					var/mob/O = M.get_ghost(TRUE)
+					if(O)
+						queue += O
+					continue
+				we++
+			for(var/datum/mind/B in G.bosses)
+				if(B.current.stat == DEAD)
+					var/mob/O = B.get_ghost(TRUE)
+					if(O)
+						queue += O
+					continue
+				we++
+		else
+			for(var/datum/mind/E in G.gangsters)
+				if(E.current.stat == DEAD)
+					continue
+				rival++
+			for(var/datum/mind/R in G.gangsters)
+				if(R.current.stat == DEAD)
+					continue
+				rival++
+	if(!we)
+		we = 1
+	if(repeat)
+		cooldown = 250+((we/(rival+we))*100)**2
+		if(!SSticker.mode.vigilantes)
+			cooldown += 650
+		addtimer(CALLBACK(src, .proc/reinforce), cooldown, TIMER_UNIQUE)
+	spawn_gangster()
+
+
+/obj/machinery/gang/backup/proc/spawn_gangster()
+	var/mob/living/carbon/human/H = new(src)
+	var/mob/dead/observer/winner
+	var/obj/item/clothing/uniform = G.inner_outfit
+	var/obj/item/clothing/suit/outerwear = new G.outer_outfit(H)
+	outerwear.armor = list(melee = 20, bullet = 35, laser = 10, energy = 10, bomb = 30, bio = 0, rad = 0, fire = 30, acid = 30)
+	outerwear.body_parts_covered = CHEST|GROIN|LEGS|ARMS
+	outerwear.desc += " Tailored for the [G.name] Gang to offer the wearer moderate protection against ballistics and physical trauma."
+	H.equip_to_slot_or_del(new uniform(H), slot_w_uniform)
+	H.equip_to_slot_or_del(outerwear, slot_wear_suit)
+	H.equip_to_slot_or_del(new /obj/item/clothing/shoes/jackboots(H), slot_shoes)
+	H.put_in_l_hand(new /obj/item/weapon/gun/ballistic/automatic/surplus/gang(H))
+	H.equip_to_slot_or_del(new /obj/item/ammo_box/magazine/m10mm/rifle(H),slot_l_store)
+	H.equip_to_slot_or_del(new /obj/item/weapon/switchblade(H),slot_r_store)
+	var/equip = SSjob.EquipRank(H, "Assistant", 1)
+	H = equip
+	if(LAZYLEN(queue))
+		var/list/mob/dead/observer/finalists = pollCandidates("Would you like to be a [G.name] gang reinforcement?", jobbanType = ROLE_GANG, poll_time = 100, ignore_category = "gang war", group = queue)
+		if(LAZYLEN(finalists))
+			winner = pick(finalists)
+	if(!winner)
+		var/list/mob/dead/observer/dead_vigils = list()
+		for(var/mob/dead/observer/O in GLOB.player_list)
+			if(!O.mind.gang_datum)
+				dead_vigils += O
+		var/list/mob/dead/observer/candidates = pollCandidates("Would you like to be a [G.name] gang reinforcement?", jobbanType = ROLE_GANG, poll_time = 100, ignore_category = "gang war", group = dead_vigils)
+		if(LAZYLEN(candidates))
+			winner = pick(candidates)
+	if(!src || !winner)
+		message_admins("No ghosts to serve as a [G.name] gang reinforement")
+		qdel(H)
+		return
+	var/datum/mind/reinforcement = new /datum/mind(winner.key)
+	reinforcement.active = 1
+	reinforcement.transfer_to(H)
```

Comment:
See the guidelines for using minds, mainly this bit: https://github.com/tgstation/tgstation/blob/master/code/datums/mind.dm#L21